jadx-core/src/main/java/jadx/core/utils/android/AndroidManifestParser.java

```diff
@@ -0,0 +1,243 @@
+		this.appStrings = parseAppStrings(appStrings);
+
+		if (isManifestFound()) {
+			checkFlags();
+			parseAttributes();
+		}
+	}
+
+	public boolean isManifestFound() {
+		return androidManifest != null;
+	}
+
+	public ApplicationParams getParseResults() {
+		return applicationParams;
+	}
+
+	@Nullable
+	public static ResourceFile getAndroidManifest(List<ResourceFile> resources) {
+		return resources.stream()
+				.filter(resourceFile -> resourceFile.getType() == ResourceType.MANIFEST)
+				.findFirst()
+				.orElse(null);
+	}
+
+	private void checkFlags() {
+		if (hasFlag(APPLICATION_LABEL) && appStrings == null) {
+			throw new IllegalArgumentException("APPLICATION_LABEL flag requires non null appStrings");
+		}
+	}
+
+	private void parseAttributes() {
+		String applicationLabel = null;
+		Integer minSdkVersion = null;
+		Integer targetSdkVersion = null;
+		Integer versionCode = null;
+		String versionName = null;
+		String mainActivity = null;
+
+		Element manifest = (Element) androidManifest.getElementsByTagName("manifest").item(0);
+		Element usesSdk = (Element) androidManifest.getElementsByTagName("uses-sdk").item(0);
+
+		if (hasFlag(APPLICATION_LABEL)) {
+			applicationLabel = getApplicationLabel();
+		}
+		if (hasFlag(MIN_SDK_VERSION)) {
+			minSdkVersion = Integer.valueOf(usesSdk.getAttribute("android:minSdkVersion"));
+		}
+		if (hasFlag(TARGET_SDK_VERSION)) {
+			String stringTargetSdk = usesSdk.getAttribute("android:targetSdkVersion");
+			if (!stringTargetSdk.isEmpty()) {
+				targetSdkVersion = Integer.valueOf(stringTargetSdk);
+			} else {
+				if (minSdkVersion == null) {
+					minSdkVersion = Integer.valueOf(usesSdk.getAttribute("android:minSdkVersion"));
+				}
+				targetSdkVersion = minSdkVersion;
+			}
+		}
+		if (hasFlag(VERSION_CODE)) {
+			versionCode = Integer.valueOf(manifest.getAttribute("android:versionCode"));
+		}
+		if (hasFlag(VERSION_NAME)) {
+			versionName = manifest.getAttribute("android:versionName");
+		}
+		if (hasFlag(MAIN_ACTIVITY)) {
+			mainActivity = getMainActivityName();
+		}
+
+		applicationParams = new ApplicationParams(applicationLabel, minSdkVersion, targetSdkVersion, versionCode,
+				versionName, mainActivity);
+	}
+
+	private String getApplicationLabel() {
+		Element application = (Element) androidManifest.getElementsByTagName("application").item(0);
+		if (application.hasAttribute("android:label")) {
+			String appLabelName = application.getAttribute("android:label");
+			if (appLabelName.startsWith("@string")) {
+				appLabelName = appLabelName.split("/")[1];
+				NodeList strings = appStrings.getElementsByTagName("string");
+
+				for (int i = 0; i < strings.getLength(); i++) {
+					String stringName = strings.item(i)
+							.getAttributes()
+							.getNamedItem("name")
+							.getNodeValue();
+
+					if (stringName.equals(appLabelName)) {
+						return strings.item(i).getTextContent();
+					}
+				}
+			} else {
+				return appLabelName;
+			}
+		}
+
+		return null;
```

Comment:
Previous implementation return "UNKNOWN" if value not found, now it returns null and cause tests to fail.